### ISSUE TITLE
[Accessibility]: Changeset list is not screen reader friendly

### ISSUE THREAD

**Issue by eseliger** [sourcegraph/sourcegraph#35009]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34089

### Problem description

The current list is implemented using CSS-grid, and the headings are just random h5s. This list should probably have attributes to make it appear as a proper list to the screen reader.

### Expected behavior

It appears as a proper list to the screen reader.

### Additional details

This will need some global effort to fix all our lists because they are all based on CSS-grid.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by courier-new** [comment]
Before we replace all our lists, we should probably do a spike on what the fix should even be. What do we need to make the list accessibility-friendly? If we want to use an actual list, could we use Wildcard's `Grid` instead of CSS grid?

**Comment by felixfbecker** [comment]
If the issue here is that in CSS grid there is no wrapping element per list item, I wonder if introducing a wrapper `<li>` element with `display: contents` would work. I've used that before in the context of flexbox.

**Comment by eseliger** [comment]
Maybe? Worth a try!

**Comment by courier-new** [comment]
🤯 I think this is exactly what we needed! What a life saver! 🙇‍♀️ I've only tested on one list so far but I'll verify the others and roll this out!

**Comment by eseliger** [comment]
Thanks Felix!! that makes this ticket so much simpler.
